miden-tx/build.rs

```diff
@@ -0,0 +1,142 @@
+use std::{collections::BTreeMap, env, fmt::Write, path::Path};
+
+use assembly::{
+    diagnostics::{IntoDiagnostic, Result},
+    Report,
+};
+use regex::Regex;
+use walkdir::WalkDir;
+
+const ASM_DIR: &str = "../miden-lib/asm";
+const ASM_TX_KERNEL_DIR: &str = "kernels/transaction";
```

Comment:
Probably we should also check the `.masm` files in the `asm/miden` and `asm/note_scripts` folders: current list of errors is not complete.

Reply:
I see, thanks! Does it still make sense to call this file and the list "kernel errors" then? The note scripts aren't really part of the kernel, so should those be exported in a separate list maybe?

Reply:
Hmm, indeed, this list contains not only kernel errors. I don't think that it is worth it to create a new list just for errors in the note scripts (there are only 7 of them, AFAIK), so probably we can just rename the list to `ERRORS` or `ASSEMBLY_ERRORS`.

Reply:
Alright, I kept it as a single error array. I called it `VM_ERRORS` now as just `ERRORS` seems to generic and it describes the context from which the errors originated from a bit better than `ASSEMBLY_ERRORS` I found.

Thanks for your comments, I addressed all of them now :)